extra_data/reader.py

```diff
@@ -72,6 +73,7 @@ class FileAccess:
     file: h5py.File
         Open h5py file object
     """
+    _fc = get_global_filecache()
```

Comment:
This runs on import, which I think is what you were trying to avoid by defining a `get_global_filecache` function in the first place. It's generally good practice to avoid running things on import, so maybe this belongs in `__init__`.

Reply:
I don't see the problem with the initialization of global variables at import time (takin into account that this code runs only once). But I agree, I did it too complex. I will create the global cache in the `filecache` module and will use it directly without additional assignment to class variable. Does it suit?